packages/backend/src/managers/apiServer.ts

```diff
@@ -0,0 +1,158 @@
+import type { Disposable } from '@podman-desktop/api';
+import type { NextFunction, Request, Response } from 'express';
+import express from 'express';
+import type { Server } from 'http';
+import * as OpenApiValidator from 'express-openapi-validator';
+import type { HttpError, OpenApiRequest } from 'express-openapi-validator/dist/framework/types';
+import path from 'node:path';
+import http from 'node:http';
+import { existsSync } from 'fs';
+import { getFreeRandomPort } from '../utils/ports';
+import * as podmanDesktopApi from '@podman-desktop/api';
+import { readFile } from 'fs/promises';
+
+const DEFAULT_PORT = 10434;
+const SHOW_API_INFO_COMMAND = 'ai-lab.show-api-info';
+
+export class ApiServer implements Disposable {
+  listener?: Server;
```

Comment:
could be private ?

Reply:
It was first, but it needs to be accessed from the unit tests

Reply:
I would rather have it private, have a `protected getListener` and have a TestApiServer exposing public the getListener.

This approach is often used in podman-desktop

https://github.com/containers/podman-desktop/blob/0b77cf2d6b70fb559abd5b4a1891a195a5d6950a/extensions/podman/scripts/podman-download.spec.ts#L59

https://github.com/containers/podman-desktop/blob/7d997640c368bb62037bed8ed337c732d78b3484/packages/main/src/tray-animate-icon.spec.ts#L27

https://github.com/containers/podman-desktop/blob/0698e4d2fd1f3fcb64bfc8d66a96003545b6ee89/packages/main/src/plugin/webview/webview-registry.spec.ts#L56